internal/forge/gitlab/edit.go

```diff
@@ -0,0 +1,52 @@
+package gitlab
+
+import (
+	"context"
+	"fmt"
+	"github.com/xanzy/go-gitlab"
+	"go.abhg.dev/gs/internal/cmputil"
+	"go.abhg.dev/gs/internal/forge"
+	"regexp"
+)
+
+const DraftRegex = `(?i)^\s*(\[Draft]|Draft:|\(Draft\))\s*`
+
+// EditChange edits an existing change in a repository.
+func (r *Repository) EditChange(_ context.Context, id forge.ChangeID, opts forge.EditChangeOptions) error {
+	if cmputil.Zero(opts) {
+		return nil // nothing to do
+	}
+
+	updateOptions := gitlab.UpdateMergeRequestOptions{}
+
+	if opts.Base != "" {
+		updateOptions.TargetBranch = &opts.Base
+	}
+
+	if opts.Draft != nil {
```

Comment:
GitLab determines the MR readiness based on the title, so we have to fetch the current title and modify it accordingly.  [Mark merge requests as drafts](https://docs.gitlab.com/ee/user/project/merge_requests/drafts.html#mark-merge-requests-as-drafts)
>Creating or editing a merge request: Add [Draft], Draft: or (Draft) to the beginning of the merge request’s title, or select Mark as draft below the Title field.

>Editing an existing merge request: Remove [Draft], Draft: or (Draft) from the beginning of the title, or clear Mark as draft below the Title field.